bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/TypedescValueImpl.java

```diff
@@ -107,6 +107,11 @@ public Object copy(Map<Object, Object> refs) {
         return this;
     }
 
+    @Override
+    public void freezeDirect() {
```

Comment:
Please create an issue to track verifying these changes with @irshadnilam's changes.